### ISSUE TITLE
Control plugin not moving the robot

### ISSUE THREAD

**Issue by MarcoMagriDev** [ADVRHumanoids/modularbots_fraunhofer#3]
### Description
I am trying to replicate this [Control plugin example](https://github.com/ADVRHumanoids/xbot2_examples/tree/master/src/homing_example). The plugin seems to be correctly loaded and executed (I started xbot2-core in verbose mode and i added a print of the `q_ref` in the `run()` method and I can see it change) but unfortunately the robot does not move.

@alaurenzi can you help me in getting this plugin work?

**Comment by alaurenzi** [comment]
Hi @MarcoMagriDev ,
**TLDR: I inadvertently changed the docker image in my last commit. You can try again pulling master first, which restores the old image.**

#### Longer story
The new image that I set by mistake inside `docker-run.sh` contains a more recent (and "experimental") version of our middleware (as seen by `xbot2-core --version` reporting a number >= `2.10.0`). This is what we'll eventually be using for CONCERT, after some more testing.

In this new version there are a few changes (pretty minor for the user), that are the cause for this issue and also #2:
  - parameter that control the activation of reference filtering have been renamed (see  [this config file](https://github.com/ADVRHumanoids/modularbots_fraunhofer/blob/32520e8e92b1eb53ba291c1947babea351cce340/ModularBot_6DOF/config/ModularBot.yaml#L48)):
    -  `/jm/tx_filter_cutoff` -> `/xbot/hal/joint_safety/filter_cutoff_hz`
    - `/jm/tx_filter_autostart` -> `/xbot/hal/joint_safety/filter_autostart`
  - each plugin needs to acquire the "hardware interfaces" it needs upon start, and this is done using `_robot->setControlMode()`; more specifically
    - override the `on_start` virtual method
    - inside `on_start`, add the line `bool ok = _robot->setControlMode(ControlMode::Position())` 
    - if resource acquisition could not be done (e.g., some other plugin is already sending positions), you usually print an error and stop, as in `if(!ok) abort();`

**Comment by MarcoMagriDev** [comment]
Sorry for the late response.

Thank you @alaurenzi, problems are solved closing issues.
